xdc-plugin/xdc.cc

```diff
@@ -186,6 +224,36 @@ struct SetProperty : public Pass {
 		bank_cell->setParam(ID(INTERNAL_VREF), RTLIL::Const(internal_vref));
 	}
 
+	void process_port_parameter(std::vector<std::string> args, RTLIL::Design* design) {
+		if (args.size() < 1) {
+			log_error("set_property: Incorrect number of arguments.\n");
+		}
+		std::string parameter(args.at(0));
+		if (args.size() < 3 || args.at(2).size() == 0) {
+			log_error("set_property %s: Incorrect number of arguments.\n", parameter.c_str());
+		}
+		std::string port_name(args.at(2));
+		std::string value(args.at(1));
+		char port[128];
+		char bit[64];
+		if (sscanf(port_name.c_str(), "%[^[]%s", port, bit) == 2) {
+			port_name = std::string(port) + " " + std::string(bit);
+		}
+		RTLIL::Module* top_module = design->top_module();
+		RTLIL::IdString port_id(RTLIL::escape_id(port_name.c_str()));
+		for (auto cell_obj : top_module->cells_) {
+			RTLIL::Cell* cell = cell_obj.second;
+			RTLIL::IdString cell_id = cell_obj.first;
+			for (auto connection : cell->connections_) {
+				RTLIL::SigSpec cell_signals = connection.second;
+				if (!strcmp(log_signal(cell_signals),port_id.c_str())) {
+					cell->setParam(RTLIL::IdString(RTLIL::escape_id(parameter)), RTLIL::Const(value));
```

Comment:
Should check here if the cell actually has the parameter of that name. For example IBUF cells does not have DRIVE and SLEW parameters

Reply:
Understood, but I don't know that if there is a generic way of checking the list of available parameters in yosys. I guess we would have to create our own map with primitives and allowed parameters.

Reply:
Added for OBUF, OBUFDS and OBUFTDS